README


This archive contains the distribution Dist-Zilla-Plugin-OnlyCorePrereqs,
version 0.020:

  Check that no prerequisites are declared that are not part of core

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.029.



Changes
Revision history for Dist-Zilla-Plugin-OnlyCorePrereqs

0.020     2015-01-03 20:57:36Z
          - re-release to fix problematic $VERSION declaration (RT#101095)

0.019     2014-12-18 06:44:28Z
          - fixed bad parsing of response from cpanmetadb, which resulted in
            invalid failures for dual-life upstream-blead modules when
            check_dual_life_versions = 0

0.018     2014-11-22 04:02:17Z
          - switch from cpanidx to cpanmetadb index

0.017     2014-08-06 05:39:32Z
          - re-release to remove README.pod from shipped dist

0.016     2014-08-01 02:13:32Z
          - fix fatal error when [MetaConfig] is used and starting_version is
            not specified (from 0.015)

0.015     2014-08-01 01:01:20Z
          - fix fatal error when used in conjunction with [MetaConfig]
            (introduced in 0.014)

0.014     2014-07-20 15:35:18Z
          - 'starting_version' now defaults to the perl prerequisite declared
            in the distribution, if it exists

0.013     2013-12-14 21:09:08Z
          - now treats dual-lifed modules that did not exist in core at all at
            the versions being checked against as the same as existing, but
            not being at a new enough version (relevant when
            check_dual_life_versions = 0)

0.012     2013-12-02 04:54:15Z
          - now also checking configure and build prereqs by default
          (RT#91081)
          - new "skip" option

0.011     2013-11-30 17:29:54Z
          - re-release to fix compile test

0.010     2013-11-11 17:02:47Z (TRIAL RELEASE)
          - fix expected string in tests for pre-5.019 perls

0.009     2013-11-10 19:40:18Z (TRIAL RELEASE)
          - more diagnostics printed in failing tests

0.008     2013-09-19 03:29:22Z
          - fix issues with check_dual_life_version feature
          - remove "check_module_versions" option

0.007     2013-09-18 00:41:19Z
          - new "check_dual_life_versions" option

0.006     2013-09-11 04:03:33Z
          - new "check_module_versions" option (removed in 0.008)

0.005     2013-09-07 00:50:54Z
          - all invalid dependencies are now reported, instead of only the
            first (David Golden, github PR #1)

0.004     2013-08-21 19:19:54Z
          - fix bad regexp causing test failure on perls < 5.019

0.003     2013-08-20 06:09:58Z
          - new config settings for 'starting_version': 'current' and 'latest'

0.002     2013-08-17 17:17:03Z
          - now respecting the deprecated_ok config
          - also check that the specific version of the prereq is available in
            the indicated perl version

0.001     2013-08-17 05:22:00Z
          - Initial release.



Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-OnlyCorePrereqs was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.32.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.010;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Check that no prerequisites are declared that are not part of core",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-OnlyCorePrereqs",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010",
  "NAME" => "Dist::Zilla::Plugin::OnlyCorePrereqs",
  "PREREQ_PM" => {
    "CPAN::DistnameInfo" => 0,
    "CPAN::Meta::Requirements" => "2.121",
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Encode" => 0,
    "HTTP::Headers" => 0,
    "HTTP::Tiny" => 0,
    "Module::CoreList" => "3.10",
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "MooseX::Types::Perl" => "0.101340",
    "YAML::Tiny" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "List::Util" => 0,
    "Moose::Util" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "if" => 0
  },
  "VERSION" => "0.020",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "CPAN::DistnameInfo" => 0,
  "CPAN::Meta::Requirements" => "2.121",
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Encode" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "HTTP::Headers" => 0,
  "HTTP::Tiny" => 0,
  "List::Util" => 0,
  "Module::Build::Tiny" => "0.039",
  "Module::CoreList" => "3.10",
  "Moose" => 0,
  "Moose::Util" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "MooseX::Types::Perl" => "0.101340",
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "YAML::Tiny" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-OnlyCorePrereqs was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-OnlyCorePrereqs was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.010;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.029.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.010"
      },
      "dist_abstract" => "Check that no prerequisites are declared that are not part of core",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-OnlyCorePrereqs",
      "dist_version" => "0.020",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::OnlyCorePrereqs",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "CPAN::DistnameInfo" => 0,
        "CPAN::Meta::Requirements" => "2.121",
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Encode" => 0,
        "HTTP::Headers" => 0,
        "HTTP::Tiny" => 0,
        "Module::CoreList" => "3.10",
        "Moose" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "MooseX::Types::Perl" => "0.101340",
        "YAML::Tiny" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.010",
        "strict" => 0,
        "version" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "List::Util" => 0,
        "Moose::Util" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "if" => 0,
        "perl" => "5.010"
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "List::Util" => 0,
      "Moose::Util" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "if" => 0,
      "perl" => "5.010"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
